ui/common/src/commonMain/kotlin/social/androiddev/common/composables/text/MastodonTextField.kt

```diff
@@ -0,0 +1,301 @@
+package social.androiddev.common.composables.text
+
+//import androidx.compose.desktop.ui.tooling.preview.Preview
+import androidx.compose.animation.animateColorAsState
+import androidx.compose.animation.core.tween
+import androidx.compose.foundation.interaction.InteractionSource
+import androidx.compose.foundation.interaction.collectIsFocusedAsState
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.shape.ZeroCornerSize
+import androidx.compose.material.ContentAlpha
+import androidx.compose.material.LocalContentAlpha
+import androidx.compose.material.LocalContentColor
+import androidx.compose.material.LocalTextStyle
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Surface
+import androidx.compose.material.TextField
+import androidx.compose.material.TextFieldColors
+import androidx.compose.material.TextFieldDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.Immutable
+import androidx.compose.runtime.State
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberUpdatedState
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.unit.dp
+import social.androiddev.common.theme.MastodonTheme
+
+@Composable
+fun MastodonTextField(
+    value: String,
+    modifier: Modifier = Modifier,
```

Comment:
Have you looked at the twitter rules for composables? I quite like it. It may be good to consider standardizing on something.

https://twitter.github.io/compose-rules/rules/

Reply:
Yes, I believe we have a detekt PR that adds this in 👍🏼

Reply:
But the modifier parameter should be always first

Reply:
If you're providing a default modifier, it should be the first in the list of defaults. Any params that don't have defaults should still be defined first. 

I'll make sure I run through all the composables I've added so far once Detekt PR is merged and fix up ordering of params where it's needed